src/chain_sync/chain_muxer.rs

```diff
@@ -850,6 +859,21 @@ where
                         self.state = ChainMuxerState::Connect(self.evaluate_network_head());
                     }
                 }
+                ChainMuxerState::Stateless(ref mut either) => match either {
+                    Either::Left(ref mut connect) => {
+                        if let Err(e) = std::task::ready!(connect.as_mut().poll(cx)) {
+                            error!("Evaluating the network head failed, retrying. Error = {e:?}");
+                        }
+                        self.state = ChainMuxerState::Stateless(Either::Right(Box::pin(
+                            tokio::time::sleep(Duration::from_secs(60)),
+                        )));
+                    }
+                    Either::Right(ref mut sleep) => {
+                        std::task::ready!(sleep.as_mut().poll(cx));
+                        self.state =
+                            ChainMuxerState::Stateless(Either::Left(self.evaluate_network_head()));
+                    }
+                },
```

Comment:
Why do you call `evaluate_network_head`?

Reply:
We need to handle or discard the inbound p2p messages. `evaluate_network_head` does the basic handling

Reply:
Done.